esp-hal/src/soc/esp32/cpu_control.rs

```diff
@@ -77,6 +77,7 @@ impl<const SIZE: usize> Stack<SIZE> {
 // is copied to the core's stack.
 static START_CORE1_FUNCTION: AtomicPtr<()> = AtomicPtr::new(core::ptr::null_mut());
 static APP_CORE_STACK_TOP: AtomicPtr<u32> = AtomicPtr::new(core::ptr::null_mut());
+static APP_CORE_STACK_GUARD: AtomicPtr<u32> = AtomicPtr::new(core::ptr::null_mut());
```

Comment:
This should be cfg-gated to not generate warnings

Reply:
ah it's not unused because of the always present `start_app_core_with_stack_guard_offset` - but I wasn't honoring `stack_guard_monitoring` in `start_app_core` 👍